x-pack/apm-server/main.go

```diff
@@ -191,12 +222,30 @@ func wrapRunServer(runServer beater.RunServerFunc) beater.RunServerFunc {
 	}
 }
 
+func closeBadger() error {
+	if badgerDB != nil {
+		return badgerDB.Close()
```

Comment:
does this access need to be coordinated to?

Reply:
Nope. `closeBadger` is only called at process exit time and never concurrently with `getBadgerDB`. I'll add a comment.